### PR TITLE
GEN-51: Run Renovate through GitHub Actions

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

With renovate being self-hosted it's possible to use a nightly version of Cargo which is required to properly generate artifacts. The hosted instance of renovate does not (and probably never will!) support nightly toolchains.

Further, we are in full control of the invocations and have access to self-hosted configurations.

## 🔍 What does this change?

- Add a workflow to run renovate every 30 min and on manual dispatch
	- The flow will authenticate with Vault and use the stored credentials for the [HASH worker bot](https://github.com/apps/hash-worker)
	- It installs Rust (the toolchain specified in `rust-toolchain.toml` and node)
	- Runs renovate
	- Stores the renovate cache (quite small) as GH artifact. In the future we might want to use S3 for that
- Add an auto-approval bot for dependencies. It will run if (and only if) the target branch is `main`, the actor (not the author of the PR) is the HASH worker, and the branch starts with `deps/`
- Allow renovate to create config-file migration PRs

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## ⚠️ Known issues

- The Rust dependency bumps still results in empty commits which is the reason why a lot of PRs are not created properly (this is a dedicated issue), but we won't get artifact-update problems with this PR
- We might need to further restrict the auto-approval constraints, but for now this should be good enough

## 🐾 Next steps

- Use S3 as Cache
- Figure out why some Rust dependencies are not bumped properly

### CODE REVIEW

File: .github/workflows/auto-approve.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    branches:
+      - main
+
+jobs:
+  auto-approve:
+    name: Dependency bump
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    if: github.actor == 'hash-worker[bot]' && startsWith(github.head_ref, 'deps/')
```
Comment (indietyp) [inline .github/workflows/auto-approve.yml:14]: This is a potential vulnerability, any branch in `deps/` will be auto-approved, would guard against that by also checking that the PR author is `hash-worker[bot]`.

**Review by indietyp — CHANGES_REQUESTED** [review]
minor concern, otherwise LGTM

File: .github/workflows/auto-approve.yml (hunk 1 above)
Comment (TimDiekmann) [reply .github/workflows/auto-approve.yml:14]: I have noted that already in the known-issues:
> We might need to further restrict the auto-approval constraints, but for now this should be good enough

The worker currently is _only_ used for creating commits/open PRs, so this is not a concern, yet. We also need to restrict the event to precisely say when an approval should happen. Ideally, only if CI is green.
```suggestion
    # Further restrict this to
    # - Check the PR author (not only the actor)
    # - Check the event (only commits should trigger auto-approvals)
    # - (optional) check for green CI
    if: github.actor == 'hash-worker[bot]' && startsWith(github.head_ref, 'deps/')
```

File: .github/workflows/auto-approve.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  auto-approve:
+    name: Dependency bump
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    # Further restrict this to
+    # - Check the PR author (not only the actor)
+    # - Check the event (only commits should trigger auto-approvals)
+    # - (optional) check for green CI
+    if: github.actor == 'hash-worker[bot]' && startsWith(github.head_ref, 'deps/')
```
Comment (TimDiekmann) [inline .github/workflows/auto-approve.yml:18]:
```suggestion
    # TODO: Further restrict this to
    # - Check the PR author (not only the actor)
    # - Check the event (only commits should trigger auto-approvals)
    # - (optional) check for green CI
    # see: https://linear.app/hash/issue/H-3313/further-restrict-auto-approval-workflow
    if: github.actor == 'hash-worker[bot]' && startsWith(github.head_ref, 'deps/')
```

File: .github/workflows/renovate.yml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        run: npm install -g renovate
+
+      - name: Run renovate
+        env:
+          LOG_LEVEL: ${{ inputs.logLevel || 'info' }}
+          RENOVATE_FORCE: ${{ github.event.inputs.overrideSchedule == 'true' && '{"schedule":null}' || '' }}
+          RENOVATE_DRY_RUN: ${{ env.dry_run == 'disabled' && 'null' || inputs.dryRun }}
+          RENOVATE_PLATFORM_COMMIT: enabled
+          RENOVATE_REPOSITORIES: ${{ github.repository }}
+          # TODO: Consider using S3 by setting `RENOVATE_REPOSITORY_CACHE_TYPE`
```
Comment (TimDiekmann) [inline .github/workflows/renovate.yml:116]:
```suggestion
          # TODO: Consider using S3 by setting `RENOVATE_REPOSITORY_CACHE_TYPE`
          # see: https://linear.app/hash/issue/H-3315/use-s3-to-store-renovate-cache
```

**Review by indietyp — APPROVED** [review]

**Pull request merged**